pkg/cloud/services/ec2/instances.go

```diff
@@ -1008,6 +1020,49 @@ func (s *Service) ModifyInstanceMetadataOptions(instanceID string, options *infr
 	return nil
 }
 
+// GetDHCPOptionSetDomainName returns the domain DNS name for the VPC from the DHCP Options.
+func (s *Service) GetDHCPOptionSetDomainName(ec2client ec2iface.EC2API, vpcID *string) *string {
+	fmt.Println("vpcID", vpcID)
+	log := s.scope.GetLogger()
+
+	if vpcID == nil {
+		log.Error(errors.New("vpcID is nil"), "vpcID is nil")
+		return nil
+	}
+
+	vpcInput := &ec2.DescribeVpcsInput{
+		VpcIds: []*string{vpcID},
+	}
+
+	vpcResult, err := ec2client.DescribeVpcs(vpcInput)
+	if err != nil {
+		return nil
+	}
+
+	dhcpInput := &ec2.DescribeDhcpOptionsInput{
+		DhcpOptionsIds: []*string{vpcResult.Vpcs[0].DhcpOptionsId},
+	}
+
+	log.Info("Fetching DHCP Options to get the domain-name", "dhcpOptionsID", *vpcResult.Vpcs[0].DhcpOptionsId)
+
+	dhcpResult, err := ec2client.DescribeDhcpOptions(dhcpInput)
```

Comment:
if this requires a new permission, you need to add it here 
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/cbf531711d117b823c587622ba0e64b94d0fd314/cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go#L78